dali/pipeline/operators/crop/crop.cc

```diff
@@ -78,43 +80,21 @@ void CropKernel(
   }
 }
 
-template<typename Out>
-DALIError_t ValidateCrop(const uint8 *in_img, int H, int W, int C, const Out *out_img) {
-  DALI_ASSERT(H > 0);
-  DALI_ASSERT(W > 0);
-  DALI_ASSERT(C == 1 || C == 3);
-  DALI_ASSERT(in_img != nullptr);
-  DALI_ASSERT(out_img != nullptr);
-  return DALISuccess;
-}
-
-template<>
-template<typename Out>
-void Crop<CPUBackend>::ValidateHelper(const Tensor<CPUBackend> *input, Tensor<CPUBackend> *output) {
-  // Validate parameters
-  DALI_CALL(ValidateCrop(
-    input->template data<uint8>(),
-    crop_[0], crop_[1], C_,
-    output->template mutable_data<Out>()));
-}
-
 template<>
 template<typename Out>
 void Crop<CPUBackend>::RunHelper(SampleWorkspace *ws, const int idx) {
   const auto &input = ws->Input<CPUBackend>(idx);
   auto output = ws->Output<CPUBackend>(idx);
 
-  // Validate
-  ValidateHelper<Out>(&input, output);
-
-  const int dataIdx = ws->thread_idx();
-  const int H = per_sample_dimensions_[dataIdx].first;
-  const int W = per_sample_dimensions_[dataIdx].second;
+  const int threadIdx = ws->thread_idx();
+  const int H = per_sample_dimensions_[threadIdx].first;
+  const int W = per_sample_dimensions_[threadIdx].second;
 
-  const int crop_y = per_sample_crop_[dataIdx].first;
-  const int crop_x = per_sample_crop_[dataIdx].second;
+  const int crop_y = per_sample_crop_[threadIdx].first;
+  const int crop_x = per_sample_crop_[threadIdx].second;
 
-  CropKernel<Out>(C_, crop_[0], crop_[1],
+  const int dataIdx = ws->data_idx();
+  CropKernel<Out>(C_, crop_height_[dataIdx], crop_width_[dataIdx],
```

Comment:
We should probably decide on storing data either per thread (threadIdx) or per sample (dataIdx). Don't know how it would influence GPU code though.

Reply:
By now it must be both, as dataIdx is useful for GPU operators that work on batch, and threadIx is useful for CPU operators. I think that the right thing to do would be to redesign the internals of crop, but by now both are needed